examples/testing-dom__download/cypress/plugins/index.js

```diff
@@ -36,6 +38,63 @@ module.exports = (on, config) => {
 
       return readXlsxFile(filename)
     },
+
+    validateZipFile (filename) {
+      // use https://github.com/cthackers/adm-zip to validate the ZIP file
+      console.log('loading zip', filename)
+      const zip = new AdmZip(filename)
+      const zipEntries = zip.getEntries()
+
+      const names = zipEntries.map((entry) => entry.entryName).sort()
+
+      console.log('zip file %s has entries %o', filename, names)
+
+      if (names.length !== 2) {
+        throw new Error(`List of files ${names.join(',')} in ${filename} has extra files`)
+      }
+
+      console.log('✅ number of entries')
+
+      if (!names.includes('a.txt')) {
+        throw new Error(`List of files ${names.join(',')} in ${filename} is missing a.txt`)
+      }
+
+      console.log('✅ has a.txt entry')
+
+      if (!names.includes('b.txt')) {
+        throw new Error(`List of files ${names.join(',')} in ${filename} is missing b.txt`)
+      }
+
+      console.log('✅ has b.txt entry')
```

Comment:
These assertions seem a little out of nowhere since I'm not familiar with the contents of the zip folder. I'm not totally sure why this is necessary. In what situation would my tests fail these checks if the zip file exists after download? Could my app have been responsible for changing the contents of the zip?

Maybe some comments would just help here explaining.

Reply:
added bunch of comments